src/handlers/wallet/exchanges/coinbase.rs

```diff
@@ -96,6 +162,46 @@ impl ExchangeProvider for CoinbaseExchange {
 }
 
 impl CoinbaseExchange {
+    fn get_api_credentials(
+        &self,
+        state: &Arc<AppState>,
+    ) -> Result<(String, String), ExchangeError> {
+        let key_name = state.config.exchanges.coinbase_key_name.clone();
+        let key_secret = state.config.exchanges.coinbase_key_secret.clone();
+
+        if key_name.is_none() || key_secret.is_none() {
+            return Err(ExchangeError::ConfigurationError(
+                "Exchange is not available".to_string(),
+            ));
+        }
+
+        Ok((key_name.unwrap(), key_secret.unwrap()))
```

Comment:
You can easily avoid `unwrap()` here:

```suggestion
        match (key_name, key_secret) {
            (Some(key_name), Some(key_secret)) => Ok((key_name, key_secret)),
            _ => Err(ExchangeError::ConfigurationError(
                "Exchange is not available".to_string(),
            ))
        }
```

Reply:
I actually completely missed this. First iteration was to get it up and running then I refactored but somehow overlooked this.

We could maybe add a warning for this as we should prob avoid it in prod?

Reply:
Sometimes I would like to add a linting error, but at the end of the day it doesn't really matter since panics are also caught on a per-request level and don't crash the entire program. Want to avoid unwraps() as a best practice, but sometimes you just need to move quickly and remove them later

Reply:
E.g. in your case here they should be fine in theory, so this is just a nitpick. Easy to fix though. Sometimes they aren't easy to fix and you just need to ship something.